sql/core/src/test/resources/sql-tests/inputs/pgSQL/with.sql

```diff
@@ -0,0 +1,1222 @@
+--
+-- Portions Copyright (c) 1996-2019, PostgreSQL Global Development Group
+--
+--
+-- WITH
+-- https://github.com/postgres/postgres/blob/REL_12_BETA2/src/test/regress/sql/with.sql
+--
+-- This test suite contains two Cartesian products without using explicit CROSS JOIN syntax.
+-- Thus, we set spark.sql.crossJoin.enabled to true.
+set spark.sql.crossJoin.enabled=true;
+
+--
+-- Tests for common table expressions (WITH query, ... SELECT ...)
+--
+
+-- Basic WITH
+WITH q1(x,y) AS (SELECT 1,2)
+SELECT * FROM q1, q1 AS q2;
+
+-- Multiple uses are evaluated only once
+-- [SPARK-19799] Support recursive SQL query
+--SELECT count(*) FROM (
+--  WITH q1(x) AS (SELECT random() FROM generate_series(1, 5))
```

Comment:
Do we have a JIRA for `generate_series`?

Reply:
Actually there is a JIRA: https://issues.apache.org/jira/browse/SPARK-27767, but is is closed as don't fix so I didn't mention it here. Shall I mention it?

Reply:
Got it. In that case, could you mention `SPARK-27767` from the beginning of this file as a `Won't Do` issue?

Reply:
Ok, done.